src/messages/formatters.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * Wraps the content inside a codeblock with no language.
+ * @param content The content to wrap.
+ * @returns The formatted content.
+ */
+export function codeBlock<C extends string>(content: C): `\`\`\`\n${C}\`\`\``;
+
+/**
+ * Wraps the content inside a codeblock with the specified language.
+ * @param language The language for the codeblock.
+ * @param content The content to wrap.
+ * @returns The formatted content.
+ */
+export function codeBlock<L extends string, C extends string>(language: L, content: C): `\`\`\`${L}\n${C}\`\`\``;
+export function codeBlock(language: string, content?: string): string {
+  return typeof content === 'undefined' ? `\`\`\`\n${language}\`\`\`` : `\`\`\`${language}\n${content}\`\`\``;
+}
+
+/**
+ * Wraps the content inside an inline code.
+ * @param content The content to wrap.
+ * @returns The formatted content.
+ */
+export function inlineCode<C extends string>(content: C): `\`${C}\`` {
+  return `\`${content}\``;
+}
+
+/**
+ * Formats the content into italic text.
+ * @param content The content to wrap.
+ * @returns The formatted content.
+ */
+export function italic<C extends string>(content: C): `_${C}_` {
+  return `_${content}_`;
+}
+
+/**
+ * Formats the content into bold text.
+ * @param content The content to wrap.
+ * @returns The formatted content.
+ */
+export function bold<C extends string>(content: C): `**${C}**` {
+  return `**${content}**`;
+}
+
+/**
+ * Formats the content into underscored text.
+ * @param content The content to wrap.
+ * @returns The formatted content.
+ */
+export function underscore<C extends string>(content: C): `__${C}__` {
+  return `__${content}__`;
+}
+
+/**
+ * Formats the content into strikethrough text.
+ * @param content The content to wrap.
+ * @returns The formatted content.
+ */
+export function strikethrough<C extends string>(content: C): `~~${C}~~` {
+  return `~~${content}~~`;
+}
+
+/**
+ * Formats the content into a quote. This needs to be at the start of the line for Discord to format it.
+ * @param content The content to wrap.
+ * @returns The formatted content.
+ */
+export function quote<C extends string>(content: C): `> ${C}` {
```

Comment:
That's a little more complicated due to the sheer amount of edge cases that'd have, for example, when you want to quote a multiline string, do you want to keep existing quotes, or flatten them? Perhaps use a specific character (`│` or `┃`) to simulate nested quotes since Discord doesn't support them?

Perhaps we want to quote only the first line, but we also want to use the simulated characters. We need some sort of composition system for this.

Reply:
Should these functions be outputting Discord compatible Markdown, or just any Markdown? Because you can do nested quotes in regular Markdown (although it's a bit messy):

> > > a
>
> > b
>
> c

```md
> > > a
>
> > b
>
> c
```

Reply:
Discord's

Reply:
Ah that's unfortunate, maybe worth having a RFC/discussion on this?

Reply:
These are _message_ formatters, not generic markdown formatters, in a respository that's part of the _discordjs_ organisation.

It would make no sense for these to support markdown that isn't compatible with Discord because these are not intended to be used as generic markdown utilities, otherwise there would be far more to include (e.g. lists, headers etc)